src/util/integerfactoring.h

```diff
@@ -0,0 +1,241 @@
+/* Include Guards */
+#ifndef INTEGERFACTORING_H
+#define INTEGERFACTORING_H
+
+
+/* Includes */
+#include <stdio.h>
+#include <stdint.h>
+#include <stdint-gcc.h>
+
+
+/* Defines */
+
+
+
+/* C++ Extern "C" Guard */
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+
+/* Data Structure Prototypes & Typedefs */
+struct GA_FACTOR_LIST;
+typedef struct GA_FACTOR_LIST GA_FACTOR_LIST;
```

Comment:
@abergeron On _which_ C++ compilers? [This Godbolt link](https://godbolt.org/g/6Fsgra) shows the pattern compiling just fine with the oldest and strictest C++ diagnostics imaginable, on ICC, GCC and Clang. [This post](https://blogs.msdn.microsoft.com/oldnewthing/20080326-00/?p=22993) by a Microsoft employee claims that this used to be required in K&R C, but C89 permits it and C++98 has always permitted it [(Example within Standard at §7.1.3(3) )](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2005/n1905.pdf). We're not targeting K&R C and even MSVC supports C89 and C++.